okhttp/src/main/java/okhttp3/RequestBody.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.Util
+import okio.BufferedSink
+import okio.ByteString
+import okio.source
+import java.io.File
+import java.io.IOException
+import kotlin.text.Charsets.UTF_8
+
+abstract class RequestBody {
+
+  /** Returns the Content-Type header for this body. */
+  abstract fun contentType(): MediaType?
+
+  /**
+   * Returns the number of bytes that will be written to `sink` in a call to [writeTo],
+   * or `-1` if that count is unknown.
+   */
+  @Throws(IOException::class)
+  open fun contentLength(): Long = -1
+
+  /** Writes the content of this request to [sink]. */
+  @Throws(IOException::class)
+  abstract fun writeTo(sink: BufferedSink)
+
+  /**
+   * A duplex request body is special in how it is **transmitted** on the network and
+   * in the **API contract** between OkHttp and the application.
+   *
+   * This method returns false unless it is overridden by a subclass.
+   *
+   * Duplex Transmission
+   * -------------------
+   *
+   * With regular HTTP calls the request always completes sending before the response may begin
+   * receiving. With duplex the request and response may be interleaved! That is, request body bytes
+   * may be sent after response headers or body bytes have been received.
+   *
+   * Though any call may be initiated as a duplex call, only web servers that are specially
+   * designed for this nonstandard interaction will use it. As of 2019-01, the only widely-used
+   * implementation of this pattern is
+   * [gRPC](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md).
```

Comment:
wanna do a labeled link at the bottom? I switched most of the codebase over to that style with the Dokka PR

Reply:
TIL you can do that in kdocs!

Reply:
I spend way too much time figuring out a regex to find all instances of these only to find that this was the only one

7ead57b